scripts/generate-genesis.py

```diff
@@ -0,0 +1,284 @@
+    shutil.move(src=monorepo_addresses_path, dst=l2_chain.addresses_path)
+
+
+def generate_combined_l1_allocs(paths, l1_chain: L1, l2_chains: list[L2]):
+    supersim_l1_allocs_paths = map(lambda l2_chain: "\"" + l2_chain.l1_allocs_path + "\"", l2_chains)
+
+    fqn = 'script/CombineAllocs.s.sol:CombineAllocs'
+    run_command([
+        'forge', 'script', fqn, "--sig", "run(string[] memory allocsPaths, string memory outputPath)", f"[{','.join(supersim_l1_allocs_paths)}]", l1_chain.combined_l1_allocs_path
+    ], cwd=paths.supersim_contracts_dir)
+
+
+def generate_l2_allocs(paths, l2_chains: list[L2]):
+  for l2_chain in l2_chains:
+
+    # Copy deploy-config.json into monorepo so that the deploy script can use it
+    monorepo_deploy_config_path = os.path.join(paths.monorepo_deploy_config_dir, f"{l2_chain.l2_chain_id}-deploy-config.json")
+    shutil.copy(src=l2_chain.deploy_config_path, dst=monorepo_deploy_config_path)
+
+    # Copy addresses.json into monorepo so that the deploy script can use it
+    monorepo_addresses_path = os.path.join(paths.monorepo_deployment_dir, f"{l2_chain.l2_chain_id}-addresses.json")
+
+    shutil.copy(src=l2_chain.addresses_path, dst=monorepo_addresses_path)
+
+    fqn = 'scripts/L2Genesis.s.sol:L2Genesis'
+    run_command([
+        'forge', 'script', fqn, "--sig", "runWithAllUpgrades()"
+    ], env={
+      'CONTRACT_ADDRESSES_PATH': monorepo_addresses_path,
+      'DEPLOY_CONFIG_PATH': monorepo_deploy_config_path,
+    }, cwd=paths.monorepo_contracts_bedrock_dir)
+
+    monorepo_forge_l2_dump_path = os.path.join(paths.monorepo_contracts_bedrock_dir, f'state-dump-{l2_chain.l2_chain_id}-fjord.json')
+
+    # Copy artifacts into supersim folders
+    shutil.move(src=monorepo_forge_l2_dump_path, dst=l2_chain.l2_allocs_path)
+    
+
+def generate_l1_genesis(paths, l1_chain: L1, l2_chains: list[L2]):
+  # we can use any of the deploy_configs, all the fields used in this process should be the same between the l2s
+  supersim_output_deploy_config_path = l2_chains[0].deploy_config_path
+
+  # this is used for verifying that the addresses have correct allocs on them
+  # TODO: validate for all l2s not just one l2
+  supersim_output_addresses = l2_chains[0].addresses_path
+
+  run_command([
+    'go', 'run', 'cmd/main.go', 'genesis', 'l1',
+    '--deploy-config', supersim_output_deploy_config_path,
+    '--l1-allocs', l1_chain.combined_l1_allocs_path,
+    '--l1-deployments', supersim_output_addresses,
+    '--outfile.l1', l1_chain.l1_genesis_path,
+  ], cwd=paths.monorepo_op_node_dir)
+
+
+def generate_l2_genesis(paths, l1_chain: L1, l2_chains: list[L2]):
+  anvil_proc = subprocess.Popen([
+    'anvil', '--silent', '--init', l1_chain.l1_genesis_path,
+  ], cwd=paths.supersim_dir)
+  
+  wait_up(8545)
+  wait_for_rpc_server('127.0.0.1:8545')
+
+  for l2_chain in l2_chains:
+    run_command([
+        'go', 'run', 'cmd/main.go', 'genesis', 'l2',
+        '--l1-rpc', 'http://127.0.0.1:8545',
```

Comment:
could be good to get the l1 rpc url or port into a variable so we can modify it at the top of the file if it ever needs to change

Reply:
fixed!